net-utils/src/lib.rs

```diff
@@ -391,22 +386,52 @@ pub fn is_host_port(string: String) -> Result<(), String> {
     parse_host_port(&string).map(|_| ())
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct SocketConfig {
-    pub reuseport: bool,
+    reuseport: bool,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    recv_buffer_size: Option<usize>,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    send_buffer_size: Option<usize>,
```

Comment:
You can add `#[derive(Copy)]` here and wouldn't need `config.clone()` elsewhere in the code.

Also, using `#[cfg(..)]` for struct fields probably isn't the best.
The fewer code differences we have across platforms is better, and here I don't think you need to change struct definition.